odoo/addons/base/models/res_lang.py

```diff
@@ -301,8 +285,8 @@ def get_installed_info_by(self, field: str) -> dict[str, LangInfo]:
         assert field in ('id', 'code', 'url_code')
         if field == 'id':
             langs = self.with_context(active_test=True).search([], order='name')
-            return tools.frozendict({lang.id: LangInfo(lang) for lang in langs})
-        return tools.frozendict({info.get(field): info for info in self.get_installed_info_by('id').values()})
+            return defaultdict(lambda: DUMMY_LANG_INFO, {lang.id: LangInfo(lang) for lang in langs})
```

Comment:
This looks like a very bad idea, since it will add the dummy lang to the dictionary!
Therefore any subsequent access to `get_installed_info_by()` will observe the extra language under all the missed keys.  Those incorrect language codes will show up in `get_installed()`, which is used for translating chart of accounts when they are installed :frowning_face: 
I recommend to return the dummy lang info in `get_info()` instead.

Reply:
Oh, you are right. I always forget about the tricky behavior of defaultdict